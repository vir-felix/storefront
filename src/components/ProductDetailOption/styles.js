export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
