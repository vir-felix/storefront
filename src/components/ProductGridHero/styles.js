export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
