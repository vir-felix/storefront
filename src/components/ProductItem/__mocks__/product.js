export default {
  product: {
    _id: "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3Q6am5hOTVrYTlhTmtNeXRteGg=",
    title: "Men's Lightweigth Synchilla",
    slug: "men's-lightweigth-synchilla",
    description: "Great jacket!",
    vendor: "Patagonia",
    isLowQuantity: false,
    isSoldOut: true,
    isBackorder: false,
    primaryImage: {
      toGrid: 1,
      priority: 0,
      productId: "cmVhY3Rpb24vcHJvZHVjdDpqbmE5NWthOWFOa015dG14aA==",
      URLs: {
        thumbnail: "/assets/files/Media/aHEWna7yeBvNFM9bL/thumbnail/25580_CICN.png",
        small: "/assets/files/Media/aHEWna7yeBvNFM9bL/small/25580_CICN.png",
        medium: "/assets/files/Media/aHEWna7yeBvNFM9bL/medium/25580_CICN.jpg",
        large: "/assets/files/Media/aHEWna7yeBvNFM9bL/large/25580_CICN.jpg"
      }
    },
    price: {
      min: 240,
      max: 240
    }
  },
  positions: [
    {
      displayWeight: 0
    }
  ]
};
