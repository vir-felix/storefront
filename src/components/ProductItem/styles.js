export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
