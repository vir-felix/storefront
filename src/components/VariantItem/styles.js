export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
