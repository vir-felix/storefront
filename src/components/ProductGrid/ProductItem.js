import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Chip from "material-ui/Chip";
import Typography from "material-ui/Typography";

const styles = (theme) => ({
  root: {},
  productInfo: {
    display: "flex",
    justifyContent: "space-between"
  },
  productMedia: {
    position: "relative"
  },
  chip: {
    borderRadius: 4,
    fontSize: 12,
    height: "auto",
    padding: theme.spacing.unit * 0.5,
    position: "absolute"
  },
  chipLabel: {
    fontWeight: theme.typography.fontWeightBold,
    padding: 0
  },
  status: {
    color: theme.palette.primary.contrastText,
    left: theme.spacing.unit,
    top: theme.spacing.unit
  },
  statusSoldOut: {
    backgroundColor: theme.palette.secondary.main
  },
  statusBackorder: {
    backgroundColor: theme.palette.secondary.dark
  },
  statusSale: {
    backgroundColor: theme.palette.error.main
  },
  statusBestseller: {
    backgroundColor: theme.palette.primary.light
  },
  warning: {
    backgroundColor: "transparent",
    ...theme.typography.caption,
    right: theme.spacing.unit,
    top: theme.spacing.unit
  },
  img: {
    height: "auto",
    width: "100%"
  }
});

@withStyles(styles)
class ProductItem extends Component {
  static propTypes = {
    classes: PropTypes.object,
    product: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    product: {}
  };

  get productStatus() {
    const { classes, product: { isBackorder, isSoldOut } } = this.props;
    let status;
    if (isSoldOut && isBackorder) {
      status = { label: "Backorder", style: `${classes.status} ${classes.statusBackorder}` };
    } else if (isSoldOut && !isBackorder) {
      status = { label: "Sold Out", style: `${classes.status} ${classes.statusSoldOut}` };
    }
    return status;
  }

  get productLowQuantity() {
    const { product: { isLowQuantity, isSoldOut } } = this.props;
    return isLowQuantity && !isSoldOut;
  }

  renderProductMedia() {
    const { classes, product: { description } } = this.props;
    const chipClasses = { root: classes.chip, label: classes.chipLabel };
    const productStatus = this.productStatus;
    console.log("product status is", this.productStatus);
    return (
      <div className={classes.productMedia}>
        {productStatus && <Chip label={productStatus.label} classes={chipClasses} className={productStatus.style} />}
        {this.productLowQuantity && <Chip label={"Low Inventory"} classes={chipClasses} className={classes.warning} />}
        <img className={classes.img} src="http://via.placeholder.com/200" alt={description} />
      </div>
    );
  }

  renderProductInfo() {
    const { classes, product: { price, title, vendor } } = this.props;
    const { range: priceRange } = price || {};
    return (
      <div className={classes.productInfo}>
        <div>
          <Typography variant="body2">{title}</Typography>
          <Typography variant="body1">{vendor}</Typography>
        </div>

        <div>
          <Typography variant="body2">${priceRange || price}</Typography>
        </div>
      </div>
    );
  }

  render() {
    return (
      <div>
        {this.renderProductMedia()}
        {this.renderProductInfo()}
      </div>
    );
  }
}

export default ProductItem;
