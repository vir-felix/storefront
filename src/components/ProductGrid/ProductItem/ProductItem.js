import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import Link from "next/link";
import { withStyles, withTheme } from "material-ui/styles";
import ButtonBase from "material-ui/ButtonBase";
import Chip from "material-ui/Chip";
import Fade from "material-ui/transitions/Fade";
import Hidden from "material-ui/Hidden";
import Typography from "material-ui/Typography";
import LoadingIcon from "mdi-material-ui/Loading";

// TODO: random number for temp images, REMOVE ONCE WE HAVE REAL DATA
const tempRandNum = 15;
const tempImgs = {
  0: {
    xs: "http://via.placeholder.com/400/E6E6E6/999999?text=FPO",
    sm: "http://via.placeholder.com/300/E6E6E6/999999?text=FPO",
    md: "http://via.placeholder.com/400/E6E6E6/999999?text=FPO",
    lg: "http://via.placeholder.com/800/E6E6E6/999999?text=FPO"
  },
  1: {
    xs: "http://via.placeholder.com/400/E6E6E6/999999?text=FPO",
    sm: "http://via.placeholder.com/600x285/E6E6E6/999999?text=FPO",
    md: "http://via.placeholder.com/700x332/E6E6E6/999999?text=FPO",
    lg: "http://via.placeholder.com/1600x780/E6E6E6/999999?text=FPO"
  },
  2: {
    xs: "http://via.placeholder.com/400/E6E6E6/999999?text=FPO",
    sm: "http://via.placeholder.com/800x248/E6E6E6/999999?text=FPO",
    md: "http://via.placeholder.com/1000x312/E6E6E6/999999?text=FPO",
    lg: "http://via.placeholder.com/1600x512/E6E6E6/999999?text=FPO"
  }
};

const styles = (theme) => ({
  [`@keyframes spin`]: {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  root: {},
  productInfo: {
    display: "flex",
    justifyContent: "space-between"
  },
  productMedia: {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  chip: {
    ...theme.typography.caption,
    borderRadius: 4,
    height: "auto",
    fontSize: 12,
    paddingBottom: theme.spacing.unit * 0.5,
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingTop: theme.spacing.unit * 0.5,
    position: "absolute",
    zIndex: 100
  },
  chipLabel: {
    fontWeight: theme.typography.fontWeightBold,
    padding: 0
  },
  status: {
    color: theme.palette.primary.contrastText,
    left: theme.spacing.unit,
    top: theme.spacing.unit
  },
  statusSoldOut: {
    backgroundColor: theme.palette.secondary.main
  },
  statusBackorder: {
    backgroundColor: theme.palette.secondary.dark
  },
  statusSale: {
    backgroundColor: theme.palette.error.main
  },
  statusBestseller: {
    backgroundColor: theme.palette.primary.light
  },
  warning: {
    backgroundColor: "transparent",
    right: theme.spacing.unit,
    top: theme.spacing.unit
  },
  img: {
    height: "auto",
    width: "100%"
  },
  imgLoading: {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  loadingIcon: {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  link: {
    ...theme.typography.body2,
    textAlign: "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});

@withStyles(styles)
@withTheme()
class ProductItem extends Component {
  static propTypes = {
    classes: PropTypes.object,
    product: PropTypes.object,
    theme: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    product: {},
    theme: {}
  };

  state = { hasImageLoaded: false };

  get productDetailHref() {
    const { product: { handle } } = this.props;
    const url = `/product/${handle}`;
    return { pathname: url };
  }

  get productStatus() {
    const { classes, product: { isBackorder, isSoldOut } } = this.props;
    let status;
    if (isSoldOut && isBackorder) {
      status = { label: "Backorder", style: `${classes.status} ${classes.statusBackorder}` };
    } else if (isSoldOut && !isBackorder) {
      status = { label: "Sold Out", style: `${classes.status} ${classes.statusSoldOut}` };
    }
    return status;
  }

  get productLowQuantity() {
    const { product: { isLowQuantity, isSoldOut } } = this.props;
    return isLowQuantity && !isSoldOut;
  }

  onImageLoad = () => {
    const { hasImageLoaded } = this.state;
    if (hasImageLoaded) return;
    this.setState({ hasImageLoaded: true });
  };

  renderProductImage() {
    const {
      classes: { img, imgLoading, loadingIcon },
      product: { description, weight },
      theme: { breakpoints: { values } }
    } = this.props;
    const { hasImageLoaded } = this.state;

    const picture = (
      <picture>
        <source srcSet={tempImgs[weight].lg} media={`(min-width: ${values.lg}px)`} />
        <source srcSet={tempImgs[weight].md} media={`(min-width: ${values.md}px)`} />
        <source srcSet={tempImgs[weight].sm} media={`(min-width: ${values.sm}px)`} />
        <img
          className={img}
          src={tempImgs[weight].xs}
          alt={description}
          onLoad={this.onImageLoad}
          ref={(image) => {
            if (image && image.complete) this.onImageLoad();
            return;
          }}
        />
      </picture>
    );

    const loading = (
      <div className={imgLoading}>
        <LoadingIcon className={loadingIcon} />
      </div>
    );

    return (
      <Fragment>
        <Fade in={hasImageLoaded}>{picture}</Fade>
        <Hidden xsUp={hasImageLoaded}>{loading}</Hidden>
      </Fragment>
    );
  }

  renderProductMedia() {
    const { classes } = this.props;
    const chipClasses = { root: classes.chip, label: classes.chipLabel };
    const { label, style } = this.productStatus || {};
    return (
      <div className={classes.productMedia}>
        {this.productStatus && <Chip label={label} classes={chipClasses} className={style} />}
        {this.productLowQuantity && <Chip label={"Low Inventory"} classes={chipClasses} className={classes.warning} />}
        {this.renderProductImage()}
      </div>
    );
  }

  renderProductInfo() {
    const { classes, product: { price, title, vendor } } = this.props;
    const { range: priceRange } = price || {};
    return (
      <div className={classes.productInfo}>
        <div>
          <Typography variant="body2">
            <Link href={this.productDetailHref}>
              <ButtonBase classes={{ root: classes.link }}>{title}</ButtonBase>
            </Link>
          </Typography>
          <Typography variant="body1">{vendor}</Typography>
        </div>

        <div>
          <Typography variant="body1">${priceRange || price}</Typography>
        </div>
      </div>
    );
  }

  render() {
    return (
      <div>
        <Link href={this.productDetailHref}>
          <a>{this.renderProductMedia()}</a>
        </Link>
        {this.renderProductInfo()}
      </div>
    );
  }
}

export default ProductItem;
