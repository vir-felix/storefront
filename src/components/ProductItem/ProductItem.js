import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Price from "@reactioncommerce/components/Price/v1";
import track from "lib/tracking/track";
import trackProductClicked from "lib/tracking/trackProductClicked";
import priceByCurrencyCode from "lib/utils/priceByCurrencyCode";
import Link from "components/Link";
import Img from "components/Img";
import BadgeOverlay from "components/BadgeOverlay";
import { styles } from "./styles";

@withStyles(styles, { withTheme: true })
@track()
class ProductItem extends Component {
  static propTypes = {
    classes: PropTypes.object,
    currencyCode: PropTypes.string.isRequired,
    product: PropTypes.object,
    theme: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    theme: {}
  };

  state = { hasImageLoaded: false };

  get productDetailHref() {
    const { product: { slug } } = this.props;
    const url = `/product/${slug}`;
    return url;
  }

  get primaryImage() {
    const { product: { primaryImage } } = this.props;
    if (!primaryImage) {
      return {
        URLs: {
          thumbnail: undefined,
          small: undefined
        }
      };
    }
    return primaryImage;
  }

  @trackProductClicked()
  handleAnchorClick = () => {}

  renderProductMedia() {
    const { classes, product: { description } } = this.props;

    return (
      <div className={classes.productMedia}>
        <Img altText={description} presrc={this.primaryImage.URLs.thumbnail} src={this.primaryImage.URLs.small} />
      </div>
    );
  }

  renderProductInfo() {
    const { classes, currencyCode, product: { pricing, title, vendor } } = this.props;
    const productPrice = priceByCurrencyCode(currencyCode, pricing) || {};

    return (
      <div >
        <div className={classes.productInfo}>
          <Typography variant="body2">
            {title}
          </Typography>
          <Typography variant="body1">
            <Price displayPrice={productPrice.displayPrice} />
          </Typography>
        </div>
        <div>
          <Typography variant="body1">{vendor}</Typography>
        </div>
      </div>
    );
  }

  render() {
    const { product } = this.props;

    return (
      <div>
        <Link
          route={this.productDetailHref}
          onAnchorClick={this.handleAnchorClick}
        >
          <BadgeOverlay product={product}>
            {this.renderProductMedia()}
            {this.renderProductInfo()}
          </BadgeOverlay>
        </Link>
      </div>
    );
  }
}

export default ProductItem;
