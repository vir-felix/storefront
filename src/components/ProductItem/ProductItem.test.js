import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "lib/theme/reactionTheme";
import product from "components/ProductDetail/__mocks__/productData.mock";
import ProductItem from "./ProductItem";

const uiStore = {
  appConfig: {
    publicRuntimeConfig: {
      externalAssetsUrl: "http://localhost:3000",
      placeholderImageUrls: {
        galleryFeatured: "/resources/placeholder.gif",
        productGrid: "/resources/placeholder.gif"
      }
    }
  }
};

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <ProductItem currencyCode="USD" product={product} uiStore={uiStore}/>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
