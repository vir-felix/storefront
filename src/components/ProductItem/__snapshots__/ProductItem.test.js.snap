// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div>
  <a
    className="WithTracking-Link--link-8"
    href="/product/example-product"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <div
      className="BadgeOverlay-badgeOverlay-9"
    >
      <div
        className="inject-ProductItem-with-uiStore-productMedia-3"
      >
        <picture
          style={
            Object {
              "opacity": 0,
              "willChange": "opacity",
            }
          }
        >
          <source
            media="(min-width: 600px)"
            srcSet="http://localhost:3000/resources/placeholder.gif"
          />
          <source
            media="(min-width: 960px)"
            srcSet="http://localhost:3000/resources/placeholder.gif"
          />
          <source
            media="(min-width: 1280px)"
            srcSet="http://localhost:3000/resources/placeholder.gif"
          />
          <img
            alt=""
            className="inject-ProductItem-with-uiStore-img-4"
            onLoad={[Function]}
            src="http://localhost:3000/resources/placeholder.gif"
          />
        </picture>
        <div
          className="inject-ProductItem-with-uiStore-imgLoading-5"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-44 inject-ProductItem-with-uiStore-loadingIcon-6"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,4V2C6.48,2 2,6.48 2,12H4C4,7.58 7.58,4 12,4Z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          className="inject-ProductItem-with-uiStore-productInfo-2"
        >
          <aside
            className="MuiTypography-root-20 MuiTypography-body2-28"
          >
            Reaction Sample Product
          </aside>
          <p
            className="MuiTypography-root-20 MuiTypography-body1-29"
          >
            $12.99 - $19.99
          </p>
        </div>
        <div>
          <p
            className="MuiTypography-root-20 MuiTypography-body1-29"
          >
            Example Manufacturer
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
